components/renku_data_services/notebooks/blueprints.py

```diff
@@ -370,10 +373,28 @@ async def _handler(
                 auth_secret = await get_auth_secret_anonymous(self.nb_config, server_name, request)
             if auth_secret.volume:
                 extra_volumes.append(auth_secret.volume)
+
+            image_pull_secret_name = None
+            if isinstance(user, AuthenticatedAPIUser) and internal_gitlab_user.access_token is not None:
+                needs_pull_secret = await requires_image_pull_secret(self.nb_config, image, internal_gitlab_user)
+
+                if needs_pull_secret:
+                    image_pull_secret_name = f"{server_name}-image-secret"
+
+                    image_secret = get_gitlab_image_pull_secret(
+                        self.nb_config, user, image_pull_secret_name, internal_gitlab_user.access_token
+                    )
+                    if image_secret:
+                        secrets_to_create.append(image_secret)
```

Comment:
`get_gitlab_image_pull_secret()` always returns a secret:

```suggestion
                    secrets_to_create.append(image_secret)
```

Reply:
sure, I have removed this conditional